src/pages/admin/Memberships/Memberships.js

```diff
@@ -0,0 +1,204 @@
+import React, { useState, useEffect } from "react";
+import { Helmet } from "react-helmet";
+import { fetchBackend } from "../../../utils";
+import { COLORS } from "../../../constants/_constants/theme";
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+import useMediaQuery from "@material-ui/core/useMediaQuery";
+import ListIcon from "@material-ui/icons/List";
+import StarBorderIcon from "@material-ui/icons/StarBorder";
+import HelpOutlineIcon from "@material-ui/icons/HelpOutline";
+import Questions from "./Questions/Questions";
+import Summary from './Summary/Summary';
+import Individual from './Individual/Individual';
+
+import {
+  List,
+  ListItem,
+  ListItemText,
+  Typography,
+  Chip,
+} from "@material-ui/core";
+
+const PERSONALIZATION_STATES = {
+  SUMMARY: {
+    index: 0,
+    displayName: "Summary",
+    icon: <StarBorderIcon fontSize="small" />,
+  },
+  QUESTION: {
+    index: 1,
+    displayName: "Question",
+    icon: <HelpOutlineIcon fontSize="small" />,
+  },
+  INDIVIDUAL: {
+    index: 2,
+    displayName: "Individual",
+    icon: <ListIcon fontSize="small" />,
+  },
+};
+
+const useStyles = makeStyles((theme) => ({
+  container: {
+    display: "flex",
+    flexDirection: "row",
+    paddingTop: "14px",
+    paddingBottom: "14px",
+  },
+  sidePanelLayout: {
+    display: "flex",
+    float: "left",
+    flexDirection: "column",
+    textAlign: "right",
+    marginRight: "3em",
+    width: "135px",
+  },
+  sidePanelButton: {
+    textAlign: "right",
+    whiteSpace: "nowrap",
+  },
+  sidePanelActiveButton: {
+    textAlign: "right",
+    whiteSpace: "nowrap",
+    borderRight: `2px solid ${COLORS.BIZTECH_GREEN}`,
+  },
+  tabsLayout: {
+    width: "80%",
+    margin: "auto",
+    [theme.breakpoints.up("sm")]: {
+      margin: "unset",
+    },
+  },
+  tabsContainer: {
+    marginBottom: "2em",
+    display: "flex",
+    flexDirection: "row",
+    [theme.breakpoints.up("sm")]: {
+      marginRight: "30px",
+    },
+  },
+  mobileFilters: {
+    overflow: "scroll",
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "space-between",
+    scrollbarWidth: "none",
+    "-ms-overflow-style": "none",
+  },
+  chipFilter: {
+    width: "100%",
+    marginRight: "0.5em",
+  },
+}));
+
+function Memberships() {
+  const [personalizationIndex, setPersonalizationIndex] = useState(
+    PERSONALIZATION_STATES.SUMMARY.index
+  );
+
+  const classes = useStyles();
+
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
+  const handlePersonalizationChange = (newIndex) => {
+    setPersonalizationIndex(newIndex);
+  };
+
+  const [membershipData, setMembershipData] = useState([]);
+
+  useEffect(() => {
+    let isSubscribed = true;
+    const getMemberships = async () => {
+      const res = await fetchBackend(`/memberships`, "GET");
+      if (isSubscribed) {
+        setMembershipData(res);
+      }
+    };
+    getMemberships();
```

Comment:
is there ever a case where isSubscribed is false? if it's set to be true on line 109. Also what does isSubscribed represent? what is it subscribed to?

Reply:
yea, it's set to false in line 118 after the call to backend is made.
there was a memory leak when setting state using fetch so setting isSubscribed to false prevents the code from setting state if the component is unmounted (i followed [this tutorial](https://juliangaramendy.dev/blog/use-promise-subscription))